apps/web/pages/week/index.page.tsx

```diff
@@ -22,24 +31,55 @@ const WeekPage = () => {
   const day = dayString ? parseISO(dayString) : new Date()
   const startDate = startOfWeek(day, { weekStartsOn: 1 })
   const endDate = endOfWeek(day, { weekStartsOn: 1 })
+  const { selectedUserId, handleUserChange, myProjectsMembersData } = useProjectMembers()
 
   const weekGridContext = useMemo(() => ({ additionalTypenames: ['Project', 'Task', 'WorkHour'] }), [])
+  const projectMemberUserId = router.query.userId?.toString()
   const [{ data: weekGridData, fetching }] = useQuery({
     query: weekGridQueryDocument,
-    variables: { from: format(startDate, 'yyyy-MM-dd'), to: format(endDate, 'yyyy-MM-dd') },
+    variables: { from: format(startDate, 'yyyy-MM-dd'), to: format(endDate, 'yyyy-MM-dd'), projectMemberUserId },
     context: weekGridContext,
   })
 
   const isDataOutdated = !!weekGridData && fetching
 
   const handleWeekChange = (newDate: Date) => {
-    router.push(isThisWeek(newDate) ? '/week' : `/week/${format(newDate, 'yyyy-MM-dd')}`)
+    const path = `/week${isThisWeek(newDate) ? '' : `/${format(newDate, 'yyyy-MM-dd')}`}${projectMemberUserId ? `?userId=${projectMemberUserId}` : ''}`
+    router.push(path)
   }
 
   return (
     <ProtectedPage>
-      <div className="flex justify-center">
-        <WeekSelector value={day} onChange={handleWeekChange} />
+      <div className="flex items-start justify-between">
+        {myProjectsMembersData.length > 0 && (
```

Comment:
Should I add this to the list box so that it looks level to the week selectors?
` <div className="mt-2">`

Reply:
Hmm, what about putting it here, in the top left quarter, would be too bloated, right?

![image](https://github.com/user-attachments/assets/d629473b-9b88-4e6a-a402-8c48643a4704)

Reply:
What do you think about this idea @moncapitaine?